### PR TITLE
Fix: set unitid when jumping to different offset

### LINKED ISSUE #4870: "Go to string number" fails to set URL properly
This bug seems to occur in master and to have existed for at least the past few months, I have hopefully nailed down the specific conditions to reproduce it, it requires a view across multiple po files (it wont be observed when translating the strings from only a single po file).

Enter a string number in the string number box and hit enter.

Pootle will navigate to that number, however it *if this number is in a different po file from where you started, it will fail to set the unit_id*
For example, I enter "25" and it sets the hash component as simply `#filter=all&offset=18`

Because it has failed to set the unit_id, if you refresh the page or attempt to use the URL pootle does not know what unit_id to go to and as such you end up at a different string number, for example 19.

The unit_id is set correctly when going from one string number within a po to another string number within the same po, for example if I click the string number box and hit enter again, because it is "navigating" within the same po file it correctly sets the url, for example `#filter=all&offset=18&unit=22`, browser refresh and using the URL then works as expected.

### PR DESCRIPTION
**Pull request by phlax**
fixes #4870

### CODE REVIEW

**Comment by unho** [comment]
Reviewed and tested. GTM

**Comment by ta2-1** [comment]
gtm

File: pootle/static/js/editor/app.js [hunk 1]
```diff
@@ -1747,19 +1747,25 @@ PTL.editor = {
...
-      this.offsetRequested = index + 1;
-      const newHash = utils.updateHashPart('offset',
-                                           this.getStartOfChunk(index),
-                                           ['unit']);
-      $.history.load(newHash);
-    } else if (index >= 0 && index <= this.units.total) {
-      const uId = this.units.uIds[(index - this.initialOffset)];
-      const newHash = utils.updateHashPart('unit', uId);
-      $.history.load(newHash);
+    index = Math.max(0, index - 1);
```
Comment (unho) [inline pootle/static/js/editor/app.js]: Is this not duplicated?

Comment (phlax) [inline pootle/static/js/editor/app.js]: yep, ill remove...

**Comment by ta2-1** [comment]
gtm again

**Comment by phlax** [comment]
landed in #d3b44e6cac

**Pull request closed without merging**